stable/prometheus-operator/templates/prometheus-operator/role.yaml

```diff
@@ -0,0 +1,79 @@
+{{- if .Values.prometheusOperator.enabled }}
```

Comment:
Could you explain why this is necessary? The existing cluster role takes care of setting RBAC requirements.

Reply:
Creating `ClusterRoles` and `ClusterRoleBindings` requires cluster-wide permission. Actually we should be able to deploy Prometheus Operator without this permission in a multi-tenancy k8s cluster. Using proper `Roles` and `RoleBindings` instead allows it.

Reply:
If you're interested in adding non-cluster-wide RBAC, then it should be done for the prometheus (and any other resources too) 

The Prometheus role is still a cluster-wide configuration and will be quite difficult to configure automatically:
1. for prometheuses running in individual namespaces, they will require RBAC to the installation namespace so they can discover alertmanager
2. prometheus to servicemonitor mapping is not obvious, so an explicit set of namespaces will probably be necessary
3. using additional scrape configs can require quite arbitrary permissions in the cluster so the user would have to configure their explicit settings in the chart

I think that adding configuration for the first two options is fine, but asking the user to configure a custom RBAC in a wrapper chart should be done for 3.

This would need good documentation about how to use it, since there are several switches that would have to be toggled in the chart in order to make full use of the options.

Reply:
I just realized that it was more tricky then expected. I'll close this PR for now.